classes/actions.php

```diff
@@ -208,10 +208,22 @@ public static function duplicate_to_course(array $modules, int $targetcourseid,
         $sourcemodinfo = get_fast_modinfo($sourcecourseid);
         $targetmodinfo = get_fast_modinfo($targetcourseid);
 
-        // If a the target section number has been specified and there is no such target section, we do nothing.
-        // This should have been prevented by the UI anyway, but we want to be extra safe here.
+        // If a new section has been specified, we create one.
         if ($sectionnum != -1 && $sectionnum > count($targetmodinfo->get_section_info_all()) - 1) {
-            return;
+            $course = $DB->get_record('course', array('id' => $targetcourseid), '*', MUST_EXIST);
+            $courseformat = course_get_format($course);
+            $courseformatoptions = $courseformat->get_format_options();
+
+            // Ensure a legitimate sectionnum is given.
+            if (isset($courseformatoptions['numsections'])) { // Course format setting.
+                $sectionnum = $courseformatoptions['numsections'] + 1;
+                // Update format setting.
+                update_course((object)array('id' => $course->id,
```

Comment:
Here we need to check for `moodle/course:update` I think

Reply:
Only possible if someone is modifying the form, and would fail anyway I believe, but checking will fail a little cleaner.